### ISSUE TITLE
reloadData does not call willDisplayCell

### ISSUE THREAD

**Issue by steventnorris-40AU** [patchthecode/JTAppleCalendar#352]
Currently, calling .reloadData() on my calendar view does not call willDisplayCell in my delegate.

In my code, I call this:

```
self.mainVw?.calVw.deselectAllDates(triggerSelectionDelegate: false)
self.mainVw?.startDate = nil
self.mainVw?.endDate = nil
self.mainVw?.calVw.reloadData()
```

Putting a breakpoint on that reload call, I then place breakpoints in my delegate method:

```
func calendar(_ calendar: JTAppleCalendarView, willDisplayCell cell: JTAppleDayCellView, date: Date, cellState: CellState) {
    let calCell = cell as! MyCalendarCell
    self.styleCalCell(calCell: calCell, cellState: cellState)
}
```
The code never reaches willDisplayCell after reloadData, which I need so my newly deselected cells will display properly without highlighting. It looks like reload should be set up to call willDisplayCell from your source, but it doesn't seem to be calling it.

**Comment by patchthecode** [comment]
Ok, this is odd. What version of the library are you using?

**Comment by steventnorris-40AU** [comment]
I am using version 6.1.5. This is the version I started with with this project. Ironically, if I call reloadDates it does hit my willDisplayCell, but not with reloadData.

**Comment by patchthecode** [comment]
ok, then  i must have had some error introduced somewhere.
Can you check the master branch and see if the error is already fixed there? 

You can test the master branch by [Following the quick instructions here](https://github.com/patchthecode/JTAppleCalendar/wiki/Common-Questions)

If it is still not fixed on master branch, then i will look into this after work.

**Comment by steventnorris-40AU** [comment]
Tried master, but it bombs on this line:

```
    /// Enables/Disables multiple selection on JTAppleCalendar
    open var allowsMultipleSelection: Bool = false {
        didSet {
            calendarView.allowsMultipleSelection = allowsMultipleSelection
        }
    }
```

Getting a nil error:

`fatal error: unexpectedly found nil while unwrapping an Optional value`

**Comment by steventnorris-40AU** [comment]
For anyone that runs across this issue while it's being fixed, the workaround is to just grabbed the selected dates or dates you want to reload prior to doing any manipulation, do your manipulation, the reload those specific dates.

```
let selectedDates = self.calVw.selectedDates
self.calVw.deselectAllDates(triggerSelectionDelegate: false)
//Other things
self.calVw.reloadDates(selectedDates)
```

**Comment by patchthecode** [comment]
Thanks for the investigation. I'll look into this after work.

**Comment by patchthecode** [comment]
if youre online now: join here -> https://gitter.im/patchthecode/JTAppleCalendar

**Comment by patchthecode** [comment]
Added the `Wont Fix` tag to this.
Will close this as it is resolved in version 7.0
Release time will be in about 1 week
